src/Tvheadend.cpp

```diff
@@ -1458,55 +1458,47 @@ PVR_ERROR CTvheadend::GetEPGForChannel
   Logger::Log(LogLevel::LEVEL_DEBUG, "get epg channel %d start %ld stop %ld", chn.iUniqueId,
            (long long)start, (long long)end);
 
+  /* Build message */
+  htsmsg_t *msg = htsmsg_create_map();
+  htsmsg_add_u32(msg, "channelId", chn.iUniqueId);
+  htsmsg_add_s64(msg, "maxTime",   end);
 
-  /* Note: Nothing to do if "async epg transfer" is enabled as all changes are pushed live to Kodi, then. */
-  if (!Settings::GetInstance().GetAsyncEpg())
```

Comment:
Maybe we should remove the non-asynchronous EPG support while we're at it and simply pass set `bSupportsAsyncEPGTransfer` to true?

Reply:
Maybe later, @Glenn-1990 reported that async epg will not work with older tvh versions. Unfortunately I forgot what version got the fix. That's why the default for async epg is still "off".

Reply:
Found it: https://github.com/tvheadend/tvheadend/commit/0de7b5c4ea478821527e66fed45c6bf059cb4e4b

Fixed for tvh > 4.0.x

Shall we drop support for 4.0.x or is this to early?

Reply:
@ksooo, it's been a while, but according to the tvh commit you picked (yes, that's the right one), it is fixed for 4.2 and un, not 4.0?